code/modules/mob/mob.dm

```diff
@@ -1356,6 +1356,12 @@ GLOBAL_LIST_INIT(slot_equipment_priority, list( \
 	SEND_SIGNAL(src, COMSIG_MOB_UPDATE_SIGHT)
 	sync_lighting_plane_alpha()
 
+/mob/proc/set_sight(datum/vision_override/O)
```

Comment:
```

/mob/proc/set_sight(datum/vision_override/O)
	QDEL_NULL(vision_type)
	if(O)
		vision_type = O
	update_sight()
```

Reply:
Wanted to get Fred to figure this one out himself but this works :P
Wouldn't it just be `set_sight(new_vision_type)` where `new_vision_type` is the type of the object?

Reply:
You could do that too, but having the args like that makes it explicitly clear what you put there.